compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5193,8 +5203,32 @@ private BSymbol getLangLibMethod(BLangInvocation iExpr, BType bType) {
         return funcSymbol;
     }
 
+    private BType createXMLNeverType(BLangInvocation iExpr, BLangXMLTextLiteral bLangXMLTextLiteral) {
+        Scope.ScopeEntry entry = symTable.rootScope.lookup(names.fromTypeKind(TypeKind.NEVER));
+        BType constraintType = (entry.symbol).type;
+        BType constrainedType = new BXMLType(constraintType, null);
+        BType type = (symTable.rootScope.lookup(names.fromTypeKind(TypeKind.XML))).symbol.type;
+        BTypeSymbol typeSymbol = type.tsymbol;
+        if (iExpr != null) {
+            constrainedType.tsymbol = Symbols.createTypeSymbol(typeSymbol.tag, typeSymbol.flags, typeSymbol.name,
+                    typeSymbol.pkgID, constrainedType, typeSymbol.owner,
+                    iExpr.pos, SOURCE);
+            return constrainedType;
+        }
+        constrainedType.tsymbol = Symbols.createTypeSymbol(typeSymbol.tag, typeSymbol.flags, typeSymbol.name,
+                typeSymbol.pkgID, constrainedType, typeSymbol.owner,
+                bLangXMLTextLiteral.pos, SOURCE);
+        return constrainedType;
+    }
+
     private void checkInvocationParamAndReturnType(BLangInvocation iExpr) {
         BType actualType = checkInvocationParam(iExpr);
+        if (actualType.tag == symTable.xmlTextType.tag) {
```

Comment:
What is the scenario we're trying to handle here?

Reply:
To handle  `xml<never> b = 'xml:createText("");` kind of scenarios.

Reply:
I don't think we should be supporting this.

Allowing this means that we are in a way calling this function and evaluating the expression at compile-time, which we do not do for function calls.

The static type of `'xml:createText(...)` at compile-time is always `'xml:Text`. So if the user wants to do this they have to say

`xml<never> b = <xml<never>> 'xml:createText("");`

Reply:
The compiler crashes for the following btw.

```ballerina
public function main(string s) {
    xml<never> b = 'xml:createText(s);
}
```

ballerina-internal.log
```log
[2020-11-20 14:22:36,824] SEVERE {b7a.log.crash} - class org.wso2.ballerinalang.compiler.tree.expressions.BLangSimpleVarRef cannot be cast to class org.wso2.ballerinalang.compiler.tree.expressions.BLangLiteral (org.wso2.ballerinalang.compiler.tree.expressions.BLangSimpleVarRef and org.wso2.ballerinalang.compiler.tree.expressions.BLangLiteral are in unnamed module of loader 'app') 
java.lang.ClassCastException: class org.wso2.ballerinalang.compiler.tree.expressions.BLangSimpleVarRef cannot be cast to class org.wso2.ballerinalang.compiler.tree.expressions.BLangLiteral (org.wso2.ballerinalang.compiler.tree.expressions.BLangSimpleVarRef and org.wso2.ballerinalang.compiler.tree.expressions.BLangLiteral are in unnamed module of loader 'app')
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkInvocationParamAndReturnType(TypeChecker.java:5283)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkFunctionInvocationExpr(TypeChecker.java:4890)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:2639)
	at org.wso2.ballerinalang.compiler.tree.expressions.BLangInvocation.accept(BLangInvocation.java:115)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:365)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:344)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:683)
	at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:53)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2957)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2925)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:2917)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:1699)
	at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:46)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2957)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2925)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeStmt(SemanticAnalyzer.java:2921)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:389)
	at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:58)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2957)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:361)
	at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:73)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2957)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2925)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:2917)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:277)
	at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:163)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyze(SemanticAnalyzer.java:251)
	at io.ballerina.projects.internal.CompilerPhaseRunner.typeCheck(CompilerPhaseRunner.java:214)
	at io.ballerina.projects.internal.CompilerPhaseRunner.performTypeCheckPhases(CompilerPhaseRunner.java:113)
	at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:325)
	at io.ballerina.projects.ModuleCompilationState$3.compile(ModuleCompilationState.java:100)
	at io.ballerina.projects.ModuleCompilationState$3.generatePlatformSpecificCode(ModuleCompilationState.java:108)
	at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:252)
	at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:118)
	at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:108)
	at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:90)
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
	at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:174)
	at io.ballerina.projects.JBallerinaBackend.from(JBallerinaBackend.java:89)
	at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:64)
	at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
	at io.ballerina.cli.cmd.RunCommand.execute(RunCommand.java:173)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at org.ballerinalang.tool.Main.main(Main.java:57)
```